### PR TITLE
added paths and mechanic

### PR DESCRIPTION
**Pull request by python3-Trying-py**
added a water path with high base mobility but no movement mechanic. added air path with double jump mechanic. added a double jump mechanic.

### CODE REVIEW

File: Scripts/AirPlayer.gd [hunk 1]
```diff
@@ -0,0 +1,16 @@
+extends "res://Scripts/Player.gd"
+
+class_name AirPath
+
+func _ready():
+	default_speed = 165
+	default_coyote_time = INF
+
+func jump():
```
Comment (urlordjames) [inline Scripts/AirPlayer.gd]: Why are we overriding this here?  This is identical to the implementation in the Player class.

File: Scripts/AirPlayer.gd [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	coyote_time = 0
+	velocity.y = 15
+	
+func djump():
```
Comment (urlordjames) [inline Scripts/AirPlayer.gd:9]: Where is this called?

File: Scripts/Player.gd [hunk 3]
```diff
@@ -100,3 +100,6 @@ func jump() -> void:
 
 func movement() -> void:
 	pass
+
+func djump() -> void:
```
Comment (urlordjames) [inline Scripts/Player.gd]: Why is this in the base class?  It is not called from the base Player.

File: Scripts/WaterPlayer.gd [hunk 4]
```diff
@@ -0,0 +1,10 @@
+extends "res://Scripts/Player.gd"
+
+class_name WaterPath
+
+func _ready():
+	default_speed = 220
+
+func jump():
```
Comment (urlordjames) [inline Scripts/WaterPlayer.gd]: Again, why are we overriding this method, it's identical to Player's implementation.  (to be fair I noticed I did this in FirePlayer too)

File: Scripts/AirPlayer.gd [hunk 5]
```diff
@@ -0,0 +1,16 @@
+extends "res://Scripts/Player.gd"
+
+class_name AirPath
+
+func _ready():
+	default_speed = 165
+	default_coyote_time = INF
```
Comment (urlordjames) [inline Scripts/AirPlayer.gd:7]: Maybe add a comment explaining why this makes a double jump work?

**Review by urlordjames — CHANGES_REQUESTED** [review]
Could you also remove jump() from FirePlayer in your next commit?  It is identical to jump() in Player, I'm not sure why I override it.

File: Scripts/AirPlayer.gd (hunk 5 above)
Comment (urlordjames) [reply Scripts/AirPlayer.gd:7]: this is a bug and will break in #2

File: Scripts/AirPlayer.gd (hunk 2 above)
Comment (python3-Trying-py) [reply Scripts/AirPlayer.gd:9]: good question

File: Scripts/AirPlayer.gd [hunk 6]
```diff
@@ -5,10 +5,6 @@ class_name AirPath
 func _ready():
 	default_speed = 165
 	default_coyote_time = INF
-
-func jump():
-	coyote_time = 0
-	velocity.y = 15
 	
 func djump():
```
Comment (urlordjames) [inline Scripts/AirPlayer.gd:9]: this still needs to go

**Review by urlordjames — CHANGES_REQUESTED** [review]
~~one last thing before I merge...~~ nevermind

**Review by urlordjames — APPROVED** [review]

**Pull request merged**